frontend/src/metabase/admin/settings/components/UploadSettings/UploadSettings.tsx

```diff
@@ -168,8 +168,8 @@ export function UploadSettingsView({
       .catch(() => showError(disableErrorMessage));
   };
 
-  const showPrefix = dbId && !showSchema;
-  const hasValidSettings = dbId && (showPrefix || schemaName);
+  const showPrefix = !!dbId;
+  const hasValidSettings = dbId && !(showSchema && !schemaName);
```

Comment:
This may be a little easier to read without the brackets
`dbId && !showSchema && schemaName`

Reply:
I don't think those are logically equivalent 😢 

A set of settings is valid if it has a dbID and either doesn't require schema or has a schema name. That expression renders all valid settings invalid because you'll never have a schema name with hidden schema.

Reply:
I think this might be a more readable version

```js
const hasValidSettings = dbId && (!showSchema || schemaName);
```

Reply:
Haha, I knew it was a mistake to rewrite that. But yeah, I think your second version is more readable